### PR TITLE
matches_number: Use long long int for 32bit + _FILE_OFFSET_BITS=64

### LINKED ISSUE #20: 1.2.3 check fails on powerpc/powerpcspe/m68k/hppa/mips
https://buildd.debian.org/status/package.php?p=isomd5sum

The only failing architecture I had ready access to is powerpc, and can reproduce.  testpyisomd5sum.py:

```
Implanting -> PASS
Implanting again w/o forcing -> PASS
Implanting again forcing -> PASS
Checking -> FAIL
Run with callback, prints offset and total
    0 - -131941381867520
    13496320 - -131941381867520
FAIL
Run with callback and abort after offset of 500000
    0 - -131941381867520
    13496320 - -131941381867520
FAIL
```

checkisomd5:

```
testiso.iso:   28d3308dc6c8e592f0587849007b738b
Fragment sums: 3d4d8dfb3959eae63e8f3c26eb464ae7edec1eab97796331d22c718ec9ed
Fragment count: 20
Supported ISO: yes
Press [Esc] to abort check.
Checking: -00.0%

The media check is complete, the result is: FAIL.

It is not recommended to use this media.
```

ISOs implanted on powerpc and checked elsewhere work fine, while ISOs implanted elsewhere and checked on powerpc fail, so it's definitely a check issue, not an implant issue.

Not sure what this could be.  All failing architectures are 32-bit userlands, but other 32-bit architectures work (i386, armhf, etc).

### PR DESCRIPTION
**Pull request by rfinnie**
Closes rhinstaller/isomd5sum#20

### CODE REVIEW

**Comment by bcl** [comment]
Thanks for the patch, I'll make sure to take a good look at this this week.

**Comment by rfinnie** [comment]
Thanks. There might be a better way to do this; I steadfastly refuse to claim I am a C programmer by any means.  Changing supported and fragmentcount to off_t seems like a hack, but does work.  (Is there something that can be passed to GCC to say "pretend *everything* is 64-bit and double longs if necessary"?)

File: libcheckisomd5.c [hunk 1]
```diff
@@ -149,7 +149,7 @@ int printMD5SUM(const char *file) {
     printf("%s:   %s\n", file, info->hashsum);
     if (strlen(info->fragmentsums) > 0 && info->fragmentcount > 0) {
         printf("Fragment sums: %s\n", info->fragmentsums);
-        printf("Fragment count: %zu\n", info->fragmentcount);
+        printf("Fragment count: %Lu\n", info->fragmentcount);
```
Comment (bcl) [inline libcheckisomd5.c]: This should be %lu off_t is signed not unsigned.

**Review by bcl — APPROVED** [review]
I think this is ok, from what I read long long int is guaranteed to be at least 64 bits on all arches.
I'm not sure it needs to be this big, but it doesn't seem to hurt anything.

**Review by bcl — CHANGES_REQUESTED** [review]
Could you rebase this on master and fix the  %Lu?

**Comment by rfinnie** [comment]
Oh, I'm very sorry I didn't follow up on this one! I can get that fix done and rebase the branch; can you re-open this please?

File: libcheckisomd5.c (hunk 1 above)
Comment (rfinnie) [reply libcheckisomd5.c:152]: Fixed

**Comment by bcl** [comment]
It won't let me reopen it, you'll need to make a new one.

**Pull request closed without merging**